addons/website_slides/wizard/base_partner_merge.py

```diff
@@ -0,0 +1,94 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+
+from itertools import groupby, islice
+
+from odoo import api, models
+from odoo.tools.misc import partition
+
+
+class MergePartnerAutomatic(models.TransientModel):
+    _inherit = 'base.partner.merge.automatic.wizard'
+
+    def _merge_internal(self, src_partners, dst_partner):
+        SlideChannelPartner_sudo = self.env['slide.channel.partner'].sudo()
+        SlidePartner_sudo = self.env['slide.slide.partner'].sudo()
+        User_sudo = self.env['res.users'].sudo()
+        all_partner_ids = (src_partners + dst_partner).ids
+        all_channels = SlideChannelPartner_sudo.search([('partner_id', 'in', all_partner_ids)]).channel_id
+        user_partner_id_lst = [(user, user.partner_id.id)
+                               for user in User_sudo.search([('partner_id', 'in', all_partner_ids)])]
+
+        # Save karma to support the karma correction after the recompute of the completion
+        init_earned_karma_by_partner_id = all_channels._get_earned_karma(all_partner_ids)
+        user_karma = {user: user.karma for (user, _) in user_partner_id_lst}
+
+        super()._merge_internal(src_partners, dst_partner)
+
+        # Recompute the completion
+        for slide_partner_relation in SlidePartner_sudo.search([('partner_id', '=', dst_partner.id)]):
+            slide_partner_relation._recompute_completion()
+        SlideChannelPartner_sudo.search([('partner_id', '=', dst_partner.id)])._recompute_completion()
+
+        # Karma correction: Karma has been impacted incorrectly by the recompute of the completion
+        earned_karma = all_channels._get_earned_karma([dst_partner.id])[dst_partner.id]
+        for user, partner_id in user_partner_id_lst:
+            delta_karma = earned_karma - init_earned_karma_by_partner_id.get(partner_id, 0)
+            user.karma = user_karma[user] + delta_karma
```

Comment:
so the karma will stay the same as before ? (even if the other partner had karma)

shouldn't we add the karma instead ? (the new karma is the sum of all karma)

Reply:
ah no ok, we will have 2 users that will share the same partner after the merge Oo
that's super weird no ? (it should be impossible to merge partner if they have a user ?)